From 882a9bb23c83f0260050f06b2c77626caf449e5a Mon Sep 17 00:00:00 2001
From: Webster Sheets <webster@web-eworks.com>
Date: Thu, 3 Jan 2019 13:05:49 -0500
Subject: [PATCH] Use dummy videodriver for modelcompiler.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d00fd1924..1512307f95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -241,7 +241,7 @@ set_target_properties(${PROJECT_NAME} modelcompiler savegamedump pioneerLib PROP
 # Optimize the models after the modelcompiler is built.
 # This really shouldn't be done inside the source tree...
 add_custom_command(TARGET modelcompiler POST_BUILD
-	COMMAND modelcompiler -b inplace
+	COMMAND ${CMAKE_COMMAND} -E env SDL_VIDEODRIVER=dummy $<TARGET_FILE:modelcompiler> -b inplace
 	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
 	COMMENT "Optimizing models" VERBATIM
 )
