From a14c7236738f8f618582ca908b1d0bc139230f68 Mon Sep 17 00:00:00 2001
From: Peter <craven@gmx.net>
Date: Wed, 6 Feb 2019 21:04:11 +0100
Subject: [PATCH 1/4] Add Get/SetDisableScreenshotInfo to LuaEngine.

---
 src/LuaEngine.cpp | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/LuaEngine.cpp b/src/LuaEngine.cpp
index 20fc8d59cc..9f37c7c003 100644
--- a/src/LuaEngine.cpp
+++ b/src/LuaEngine.cpp
@@ -331,6 +331,21 @@ static int l_engine_set_fullscreen(lua_State *l)
 	return 0;
 }
 
+static int l_engine_get_disable_screenshot_info(lua_State *l)
+{
+	LuaPush<bool>(l, Pi::config->Int("DisableScreenshotInfo") != 0);
+	return 1;
+}
+
+static int l_engine_set_disable_screenshot_info(lua_State *l)
+{
+	if (lua_isnone(l, 1))
+		return luaL_error(l, "SetDisableScreenshotInfo takes one boolean argument");
+	const bool disable = LuaPull<bool>(l, 1);
+	Pi::config->SetInt("DisableScreenshotInfo", (disable ? 1 : 0));
+	Pi::config->Save();
+	return 0;
+}
 static int l_engine_get_vsync_enabled(lua_State *l)
 {
 	lua_pushboolean(l, Pi::config->Int("VSync") != 0);
@@ -1099,6 +1114,8 @@ void LuaEngine::Register()
 		{ "SetVideoResolution", l_engine_set_video_resolution },
 		{ "GetFullscreen", l_engine_get_fullscreen },
 		{ "SetFullscreen", l_engine_set_fullscreen },
+		{ "GetDisableScreenshotInfo", l_engine_get_disable_screenshot_info },
+		{ "SetDisableScreenshotInfo", l_engine_set_disable_screenshot_info },
 		{ "GetVSyncEnabled", l_engine_get_vsync_enabled },
 		{ "SetVSyncEnabled", l_engine_set_vsync_enabled },
 		{ "GetTextureCompressionEnabled", l_engine_get_texture_compression_enabled },

From 21f85dad7c5686d931598e4de94b76cd0ab15683 Mon Sep 17 00:00:00 2001
From: Peter <craven@gmx.net>
Date: Wed, 6 Feb 2019 21:04:39 +0100
Subject: [PATCH 2/4] Add DisableScreenshotInfo to settings window.

---
 data/lang/ui-core/en.json              | 8 ++++++++
 data/pigui/modules/settings-window.lua | 6 ++++++
 2 files changed, 14 insertions(+)

diff --git a/data/lang/ui-core/en.json b/data/lang/ui-core/en.json
index e4ad5882f5..81ed8379b3 100644
--- a/data/lang/ui-core/en.json
+++ b/data/lang/ui-core/en.json
@@ -283,6 +283,14 @@
     "description": "tooltip description in settings menu",
     "message": "Lower gives better performance, at cost of visual aesthetics"
   },
+  "DISABLE_SCREENSHOT_INFO": {
+    "description": "",
+    "message": "Disable Screenshot Info"
+  },
+  "DISABLE_SCREENSHOT_INFO_DESC": {
+    "description": "",
+    "message": "Hide all displayed info, including system coordinates, when in screenshot mode (toggle HUD mode)."
+  },
   "DISMISS": {
     "description": "Terminate crew memeber's contract",
     "message": "Dismiss"
diff --git a/data/pigui/modules/settings-window.lua b/data/pigui/modules/settings-window.lua
index 81d65d394f..602003825e 100644
--- a/data/pigui/modules/settings-window.lua
+++ b/data/pigui/modules/settings-window.lua
@@ -161,6 +161,7 @@ local function showVideoOptions()
 
 	local textCompress = Engine.GetTextureCompressionEnabled()
 	local gpuJobs = Engine.GetGpuJobsEnabled()
+	local disableScreenshotInfo = Engine.GetDisableScreenshotInfo()
 
 	local fractalDetail = keyOf(detailLabels,keyOf(detailLevels, Engine.GetFractalDetailLevel()))-1
 	local cityDetail = keyOf(detailLabels,keyOf(detailLevels, Engine.GetCityDetailLevel()))-1
@@ -211,6 +212,11 @@ local function showVideoOptions()
 	if c then
 		Engine.SetGpuJobsEnabled(gpuJobs)
 	end
+	ui.nextColumn()
+	c,disableScreenshotInfo = checkbox(lui.DISABLE_SCREENSHOT_INFO, disableScreenshotInfo, lui.DISABLE_SCREENSHOT_INFO_DESC)
+	if c then
+		Engine.SetDisableScreenshotInfo(disableScreenshotInfo)
+	end
 	ui.columns(1,"",false)
 
 

From 9537e821d2c3b1870295be11b50733b6d97c11bc Mon Sep 17 00:00:00 2001
From: Peter <craven@gmx.net>
Date: Wed, 6 Feb 2019 21:05:02 +0100
Subject: [PATCH 3/4] Show screenshot info from lua.

---
 data/pigui/game.lua | 40 +++++++++++++++++++++++++++++-----------
 1 file changed, 29 insertions(+), 11 deletions(-)

diff --git a/data/pigui/game.lua b/data/pigui/game.lua
index a4ecf3166b..c17f8be73a 100644
--- a/data/pigui/game.lua
+++ b/data/pigui/game.lua
@@ -934,6 +934,20 @@ local function displayOnScreenObjects()
 	end
 end
 
+local function displayScreenshotInfo()
+	if not Engine.GetDisableScreenshotInfo() then
+		local current_system = Game.system
+		if current_system then
+			local current_path = current_system.path
+			local frame = player.frameBody
+			if frame then
+				local info = frame.label .. ", " .. current_system.name .. " (" .. current_path.sectorX .. ", " .. current_path.sectorY .. ", " .. current_path.sectorZ .. ")"
+				ui.addStyledText(Vector(20, 20), ui.anchor.left, ui.anchor.top, info , colors.white, pionillium.large)
+			end
+		end
+	end
+end
+
 ui.registerHandler('game', function(delta_t)
 		-- delta_t is ignored for now
 		player = Game.player
@@ -944,18 +958,22 @@ ui.registerHandler('game', function(delta_t)
 		ui.withStyleColors({ ["WindowBg"] = colors.transparent }, function()
 			ui.window("HUD", {"NoTitleBar", "NoResize", "NoMove", "NoInputs", "NoSavedSettings", "NoFocusOnAppearing", "NoBringToFrontOnFocus"}, function()
 				center = Vector(ui.screenWidth / 2, ui.screenHeight / 2)
-				if Game.CurrentView() == "world" and ui.shouldDrawUI() then
-					if Game.InHyperspace() then
-						displayHyperspace()
-						callModules("hyperspace")
+				if Game.CurrentView() == "world" then
+					if ui.shouldDrawUI() then
+						if Game.InHyperspace() then
+							displayHyperspace()
+							callModules("hyperspace")
+						else
+							displayOnScreenObjects()
+							displayReticule()
+							ui.displayPlayerGauges()
+							displayTargetScanner()
+							displayHyperspaceCountdown()
+							callModules("game")
+							ui.radialMenu("worldloopworld")
+						end
 					else
-						displayOnScreenObjects()
-						displayReticule()
-						ui.displayPlayerGauges()
-						displayTargetScanner()
-						displayHyperspaceCountdown()
-						callModules("game")
-						ui.radialMenu("worldloopworld")
+						displayScreenshotInfo()
 					end
 				else
 					if ui.shouldDrawUI() then

From 1db4276f38b7ec754819b15c74832fc651cdcaae Mon Sep 17 00:00:00 2001
From: Peter <craven@gmx.net>
Date: Wed, 6 Feb 2019 21:05:13 +0100
Subject: [PATCH 4/4] Remove screenshot info from C++

---
 src/Pi.cpp | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/Pi.cpp b/src/Pi.cpp
index d1a28ea66e..1c2c36ecc6 100644
--- a/src/Pi.cpp
+++ b/src/Pi.cpp
@@ -1457,24 +1457,6 @@ void Pi::MainLoop()
 		Pi::renderer->ClearDepthBuffer();
 		if (DrawGUI) {
 			Gui::Draw();
-		} else if (game && game->IsNormalSpace()) {
-			if (config->Int("DisableScreenshotInfo") == 0) {
-				const RefCountedPtr<StarSystem> sys = game->GetSpace()->GetStarSystem();
-				const SystemPath sp = sys->GetPath();
-				std::ostringstream pathStr;
-
-				// fill in pathStr from sp values and sys->GetName()
-				static const std::string comma(", ");
-				pathStr << Pi::player->GetFrame()->GetLabel() << comma << sys->GetName() << " (" << sp.sectorX << comma << sp.sectorY << comma << sp.sectorZ << ")";
-
-				// display pathStr
-				Gui::Screen::EnterOrtho();
-				Gui::Screen::PushFont("ConsoleFont");
-				static RefCountedPtr<Graphics::VertexBuffer> s_pathvb;
-				Gui::Screen::RenderStringBuffer(s_pathvb, pathStr.str(), 0, 0);
-				Gui::Screen::PopFont();
-				Gui::Screen::LeaveOrtho();
-			}
 		}
 
 		// XXX don't draw the UI during death obviously a hack, and still
