From 2ab957f5086f0febb8d5f44dfa06794b7c207388 Mon Sep 17 00:00:00 2001
From: Webster Sheets <webster@web-eworks.com>
Date: Fri, 29 Mar 2019 13:25:11 -0400
Subject: [PATCH] Fix an error preventing keybinding when joysticks are
 disabled.

---
 src/LuaPiGui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/LuaPiGui.cpp b/src/LuaPiGui.cpp
index 5b041ed6d6..2f9ca40692 100644
--- a/src/LuaPiGui.cpp
+++ b/src/LuaPiGui.cpp
@@ -822,7 +822,7 @@ static int l_pigui_get_keybinding(lua_State *l)
 			}
 			if (binding.compare("")) break;
 		}
-	} else {
+	} else if (key != 0) {
 		// hard coding is bad, but is instantiating a keybinding every frame worse?
 		binding = "Key" + std::to_string(key);
 		if (mod > 0) binding += "Mod" + std::to_string(mod);
